### PR TITLE
Show progress in the notification area when download codegen cli

### PR DESCRIPTION
**Pull request by dooriya**

### CODE REVIEW

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 1]
```diff
@@ -580,65 +580,58 @@ export class CodeGeneratorCore {
...
+        channel,
+        `Step 1: Downloading ${DigitalTwinConstants.codeGenCli} v${
+            newVersion} package ...`);
+    const downloadOption: request
+        .OptionsWithUri = {method: 'GET', uri: packageUri, encoding: null};
+    const zipData = await request(downloadOption).promise();
+    const tempPath = path.join(os.tmpdir(), FileNames.iotworkbenchTempFolder);
+    const filePath = path.join(tempPath, `${md5value}.zip`);
+    fs.writeFileSync(filePath, zipData);
+    // clearInterval(loading);
```
Comment (eriolchan) [inline src/DigitalTwin/CodeGeneratorCore.ts]: better to remove unused code here

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 2]
```diff
@@ -696,7 +689,22 @@ export class CodeGeneratorCore {
...
         {location: vscode.ProgressLocation.Notification, title: processTitle},
-        async () => {
+        async (progress) => {
+          progress.report({increment: 0});
+
+          setTimeout(() => {
+            progress.report({increment: 10, message: `still going...`});
+          }, 1000);
+
+          setTimeout(() => {
```
Comment (eriolchan) [inline src/DigitalTwin/CodeGeneratorCore.ts]: why need to report 40%?

File: src/DigitalTwin/CodeGeneratorCore.ts (hunk 1 above)
Comment (dooriya) [reply src/DigitalTwin/CodeGeneratorCore.ts]: Good catch. Will remove it in next iteration

---
In reply to: [361902534](https://github.com/microsoft/vscode-iot-workbench/pull/894#discussion_r361902534) [](ancestors = 361902534)

File: src/DigitalTwin/CodeGeneratorCore.ts (hunk 2 above)
Comment (dooriya) [reply src/DigitalTwin/CodeGeneratorCore.ts]: Show the progress of the download pregress of the CodeGen CLI, Similar to [https://github.com/microsoft/vscode-extension-samples/tree/master/progress-sample](https://github.com/microsoft/vscode-extension-samples/tree/master/progress-sample)

---
In reply to: [361902619](https://github.com/microsoft/vscode-iot-workbench/pull/894#discussion_r361902619) [](ancestors = 361902619)

File: src/DigitalTwin/CodeGeneratorCore.ts [hunk 3]
```diff
@@ -580,65 +580,58 @@ export class CodeGeneratorCore {
...
-
-      // Verify
-      // Validate hash code
-      utils.channelShowAndAppend(
-          channel, 'Step 2: Validating hash code for the package ...');
-      const hashvalue = await FileUtility.getFileHash(filePath);
-      if (hashvalue !== md5value) {
+          `the downloaded ${DigitalTwinConstants.codeGenCli} v${
+              newVersion} package has been corrupted.`);
+      if (installOrUpgrade === 1) {
```
Comment (dilin-MS) [inline src/DigitalTwin/CodeGeneratorCore.ts]: What does number 1 means here? Better to use CodeGenCliOperation enum value to better illustrate the meaning.

**Pull request merged**